.github/workflows/release.yml

```diff
@@ -0,0 +1,63 @@
+name: Release
+on:
+  release:
+    types:
+      # works for building pre-releases
+      - published
+      # works for releases and changing from pre-release to release
+      - released
+
+env: 
+ 
+  DOCKER_IMAGE: newrelic/nri-kube-events
+  RELEASE_VERSION: ${{ github.event.release.tag_name }}
+  S3_PATH: s3://nr-downloads-main/infrastructure_agent
+  PRERELEASE: ${{ github.event.release.prerelease }}
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Extract version from release tag
+        run: |
+          echo "RELEASE_VERSION=$(echo $RELEASE_VERSION | grep -oE '([0-9.]*[0-9])$')" >> $GITHUB_ENV
+
+      - name: Test for prerelase
+        if: ${{ github.event.release.prerelease }}
```

Comment:
have you tested this? i had some issues trying to have this working , i had to use something like this `if: github.event.action == 'prereleased'`. Although how you implemented ,is clearer

Reply:
It took some time to figure out (the docs are a bit obscure), but `github.event` has the same content as the corresponding object from [webhook payloads](https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#release), which includes the `release` field and the `prerelease` bool.

In any case I have tested this on my local forks and seems to work well.